src/main/java/org/spongepowered/common/mixin/core/world/MixinWorldServer.java

```diff
@@ -75,4 +84,37 @@ public void onPostInit(CallbackInfoReturnable<World> ci) {
         IMixinWorld world = (IMixinWorld) ci.getReturnValue();
         world.updateWorldGenerator();
     }
+
+    @Overwrite
+    public void updateAllPlayersSleepingFlag() {
+        this.allPlayersSleeping = false;
+        if (!this.playerEntities.isEmpty()) {
+            int ignoredPlayers = 0;
+            int sleepingPlayers = 0;
+            for (EntityPlayer entityPlayer : this.playerEntities) {
+                if (entityPlayer.isSpectator()
+                        || (entityPlayer instanceof Human && ((Human)entityPlayer).isSleepingIgnored())) {
```

Comment:
@simon816 My reasoning here was that `EntityPlayerMP` both extends `EntityPlayer` and implements `Human`, and would therefore be picked up by this comparison. However, given the comment on the API PR, I'll change it accordingly.